src/widgets/table/table.rs

```diff
@@ -233,6 +250,11 @@ impl<'a> Table<'a> {
     /// ```
     pub fn new<R, C>(rows: R, widths: C) -> Self
     where
+        // The `Table::new` function could be made more flexible by accepting `R: IntoIterator`,
+        // with `R::Item: Into<Row<'a>>`.  This may be desirable at some point, but would be a
+        // breaking change, as type inference could no longer use `Table::new` to infer the type of
+        // elements of `R`.  For example, `Table::new(vec![], [Constraint::Min(10)])` would not
+        // compile, because the type of `vec![]` would be unknown.
```

Comment:
I think we might want to break the empty vec use case. I don't recall why we didn't already.
The caller can use `Table::default().widths([Constraint::Min(10)])` for that.

This would imply an updated commit message and an update in the breaking changes doc.

Reply:
Agreed.  The empty vec/array case seems like it is primarily used in the test cases, but an empty table wouldn't show up as often in actual usage.

Updated here, and in the breaking changes doc.